### ISSUE TITLE
SearchRequest breaks any request using "+"

### ISSUE THREAD

**Issue by hypixus** [JohnnyCrazy/SpotifyAPI-NET#771]
**tl;dr: + in search request turns into %2b, making spotify API not understand it properly.**

Hello,

I'm trying to use this library for a simple Discord RPC project, and I noticed a very specific behavior.
Whenever I'm calling Search API, as documented on Spotify docs [here](https://developer.spotify.com/console/get-search-item/), using additional query information, such as `+artist:X` query automatically translates the plus into "htmlified" plus, namely `%2b`.

Problem is, Spotify API itself does not understand such encoding, and instead of reacting to(example):

`https://api.spotify.com/v1/search?type=album&limit=50&offset=0&query=Reprise+artist%3AFeint`

throws an empty search result, because what actually library sends is:

`https://api.spotify.com/v1/search?type=album&limit=50&offset=0&query=Reprise%2Bartist%3AFeint`.

This is the code that makes a request:

![image](https://user-images.githubusercontent.com/13383985/170546743-8be20522-6760-40a3-bfb6-8cdf851ca66a.png)

Here is the caught request in Visual Studio's diagnostic tools.

![image](https://user-images.githubusercontent.com/13383985/170545689-9c777db5-9804-44c4-93aa-493524a74ddd.png)

Proof that + and %2b are treated differently in Postman, both are using a fresh token to ensure API is not at fault:

_Version with `%2b`_
![image](https://user-images.githubusercontent.com/13383985/170545823-c5706935-06c2-48e1-9580-a8f53338c314.png)

_Version with `+`_
![image](https://user-images.githubusercontent.com/13383985/170545886-84ccb5fe-3f30-4473-81ef-9bd2fd86976d.png)

**Comment by JohnnyCrazy** [comment]
Hi,

Thanks for the detailed report. It seems like even their web console doesnt support it correctly...

https://developer.spotify.com/console/get-search-item/?q=remaster%2520track%3ADoxy%2Bartist%3AMiles%2520Davis&type=track%2Cartist&market=ES&limit=10&offset=5&include_external=

Can you try replacing '+' with a normal space? It will get replaced by `%20`, which the spotify API seems to understand. But not sure if the actual response data matches the query...maybe you can check that

**Comment by JohnnyCrazy** [comment]
Looking at the response in more detail, it seems like `%20` is not treated correctly as a seperator...

I'm gonna check if we can disable `+` encoding, as it's normally not necessary at all

**Comment by hypixus** [comment]
Yeah, this is rather ridiculous... There is a reason why most people I know that work with this API aren't exactly in love with it. Sigh.

As to why I used plus - the aforementioned console had it as an example to make the queries more precise. Ironic.

In case of any further assistance is required I'll gladly help.

**Comment by cdowideit** [comment]
It seems to work well for me with a simple blank. However, I have a related question: How can I specify e.g. market, but also offset and limit, if at all?

**Comment by JohnnyCrazy** [comment]
You add it to the search request object:

```csharp
const request = new SearchRequest(SearchRequest.Types.Album, "Muse");
request.Market = "DE";
request.Limit = 50;
request.Offset = 10;
// ...
```

Next time, please open a seperate discussion, thanks!

**Comment by JohnnyCrazy** [comment]
If I read the [latest documentation](https://developer.spotify.com/documentation/web-api/reference/search?q=remaster%2520track%3ADoxy%2Bartist%3AMiles%2520Davis&type=track%2Cartist&market=ES&limit=10&offset=5&include_external=) correctly, they switched to using spaces (`%20`) as separator. This should work correctly with our encoding. Let me know if it doesn't!
